### ISSUE TITLE
Maintenance

### ISSUE THREAD

**Issue by tarwich** [quilt-js/js-xlsx#1]
We need to figure out how we're going to handle maintenance. This might need broken out into multiple issues.

## Testing

`js-xlsx` had a lot of unit tests. When I ran them myself there were over 100. However, they didn't pass even on the `master` branch. We should either fix the existing tests or use a different system. Personally I'm in love with wercker.com.

## Scope creep

We should figure out what constitutes "_in scope_" and find a way to accept plugins or something for everything else.

## Name

I kept the name js-xlsx, but it confuses me being in a repo called node-xlsx. Should we rename it before it gets bookmarked too much?

## Protobi

Are we going to merge his stuff in? I thought someone said he's keeping it up to date, so it should merge. It's just a question of _testing_ and _scope creep_.

**Comment by eddyparkinson** [comment]
## Testing
wercker.com sounds fine. I have not used automated testing for a while, so not sure what is good.

## Scope creep
Latest version of MS Excel vs Apache OpenOffice Calc? Is there a way to accept both. I would be tempted to accept both, but when there is a conflict which direction would you jump? There is demand for excel and  OpenOffice is available to all.

## Name
Can you suggest some options and then pick on or after Dec 28. Hope this is enough time.
Rename Repo: SheetJS-Org

## Protobi
Yes, use Protobi as the starting point - unless you spot something better, but looks to be the best.
I invited @pietersv as admin - he is an expert on this repo

**Comment by tarwich** [comment]
## Testing

I'll see if there are any platforms that will allow automation of applications. I know you can automate Windows, so you could conceivably have a testing system that opened worksheets and tested to see if the correct value(s) were in the cell(s). Might be overkill.

## Name

I'm a sucker for names that have literary merit. For example, since this is an amalgamation of a ton of different forks, it's more like a bunch of `sheet-js` sewn together. That would be... `quilt-js`?

**Comment by protobi-pieter** [comment]
@tarwich et al --  Thanks for creating momentum going on this great resource.

When creating the styles branch, I was able to successfully run all the unit tests in the browser and in node 0.12, and recall it was something like 30,000+ tests in the browser and 20,000+ tests in Node.js.   Being unable to run the test suite on Node 4+ has been my main hurdle.

One good starting point would be to identify a fork that's minimally different from the SheetJS/js-xlsx root, but which gets the tests running.  I can then port the styles extensions into a fork of that.  

Re automating tests, it would be really powerful to actually test the application in Microsoft Office and Libre Office on Windows, OS X and Linux in different locales. If that's not feasible, one advantage of the SheetJS/js-xlsx library is that is both read-write and can thus be the engine the reads the output to verify the cell contents and other aspects.

**Comment by protobi-pieter** [comment]
Great news that the main project now supports Node 6.0 (https://github.com/SheetJS/js-xlsx/issues/513)!  

I think a good path may be to start clean from the latest version of the main project.  I can clone that master branch, verify the tests, and port over styles changes.   This will allow me the benefit of squashing prior changes and a second chance to get the upstream, #beta and master branches right.

Now that the root project is back active, from there I wonder if the simplest approach is to submit most changes as focused PR requests to the root project, keep each PR associated with a specific issue, and with appropriate unit tests.

The styles branch protobi/js-pptx  can pull changes from the root project periodically, and accept new issues/PRs related to related to read/write of styles.

As an aside, I see a recent commit on this repo https://github.com/node-xlsx/js-xlsx had changes directly to the `xlsx.js` file at the top-level.  This file should only be generated automatically by the `make` process, and all changes made to the `/bits/*.js` files.   Following the make and testing process would be really important for all PRs.

**Comment by eddyparkinson** [comment]
Name
---
@tarwich I like your idea "quilt-js -  bunch of sheet-js sewn together" suggest we go with that.

**Comment by tarwich** [comment]
@pietersv Yeah... We should probably blow things away here and start from SheetJS. We could keep a branch like `master` or `dev` as a branch that has our features already merged in regardless of whether SheetJS has accepted them.

**Comment by eddyparkinson** [comment]
Renamed organization to "https://github.com/quilt-js"

**Comment by eddyparkinson** [comment]
- Replaced quilt-js/js-xlsx with sheetjs/js-xlsx.
- Added formulas https://github.com/SheetJS/js-xlsx/pull/307 to bits and re-generated xlsx.js

@pietersv how to setup unit tests? Thoughts on where to start.
